src/dosext/mfs/lfn.c

```diff
@@ -1328,6 +1328,11 @@ int mfs_lfn(void)
 {
 	int carry, ret;
 
+	if (!mach_fs_enabled) {
+		CARRY;
+		return 0;
+	}
+
```

Comment:
DR-DOS < 7 is unsupported by MFS as that currently needs the CDSPtr in LOL, so there are no further MFS requests on that version of DOS, hence no crashing from there. But LFN calls are always possible due to the revectoring that is setup unconditionally. LFN attempts to use CDSptr from LOL during QUALIFY_FILENAME. I may be missing something but why does LFN need to be activated if we have no MFS drives? I wondered about it before and I think you mentioned VFAT, but surely VFAT version of QUALIFY_FILENAME shouldn't care about remote drives. It may be naive, but I had figured that QUALIFY_FILENAME would only provide answers for the drives it owns, i.e. MFS and no others.